bin/describe.ml

```diff
@@ -152,8 +175,80 @@ module Crawl = struct
     Digest.generic
       (Lib.name lib, Path.to_string (Lib_info.src_dir (Lib.info lib)))
 
+  (** Module that computes the direct dependencies between compilation units *)
+  module Deps : sig
+    val read :
+         options:options
+      -> use_pp:bool
+      -> obj_dir:Path.Build.t Obj_dir.t
+      -> modules:Modules.t
+      -> Module.t
+      -> Module.t list Action_builder.t Memo.build
+  end = struct
+    (** Reads the dependencies of the compilation unit [unit] (for its
+        implementation and its interface, if any) *)
+    let read_deps_of_unit ~obj_dir ~modules unit =
+      let open Action_builder.O in
+      let+ deps_for_impl =
+        (* compute the dependencies for the implementation part *)
+        let ml_kind = Dune_util.Ml_kind.Impl in
+        Ocamldep.read_immediate_deps_of ~obj_dir ~modules ~ml_kind unit
+      and+ deps_for_intf =
+        (* compute the dependencies for the interface part *)
+        let ml_kind = Dune_util.Ml_kind.Intf in
+        Ocamldep.read_immediate_deps_of ~obj_dir ~modules ~ml_kind unit
+      in
+      let open Module.Obj_map in
+      (* return the union of the dependencies, as a sorted list with no
+         duplicates *)
+      union
+        ~f:(fun _ () () -> Some ())
+        (of_list_unit deps_for_impl)
+        (of_list_unit deps_for_intf)
```

Comment:
Do you think it's perhaps better to print these sets separately and let callers merge them if needed?

Reply:
Good idea. Implemented in the recently pushed changes.